relay/relay.go

```
@@ -27,6 +27,22 @@ import (
+// the entire map on every eviction attempt.
+const evictionSampleSize = 64
+
+// Serializes UDP destination connection cap-check and creation to avoid
+// concurrent goroutines temporarily exceeding maxUDPConnections.
+var udpConnCreateMu sync.Mutex
+
+func releasePacketBuffer(buf []byte) {
+	if buf == nil || cap(buf) == 0 {
+		return
+	}
+	bufferPool.Put(buf[:cap(buf)])
```

golangci-lint (2.12.2)

[error] SA6002: argument should be pointer-like to avoid allocations

(staticcheck)